sheets/mail_api.py

```diff
@@ -1,21 +1,57 @@
 """Mail API for sheets app"""
+import logging
 from urllib.parse import urlencode
 from collections import namedtuple
 
 import requests
 from django.conf import settings
+from rest_framework import status
 
 from ecommerce.constants import BULK_ENROLLMENT_EMAIL_TAG
 from mail.constants import MAILGUN_API_DOMAIN
+from sheets.constants import MAILGUN_API_TIMEOUT_RETRIES
 from sheets.utils import format_datetime_for_mailgun
 from mitxpro.utils import has_all_keys
 
+
+log = logging.getLogger(__name__)
+
 BulkAssignmentMessage = namedtuple(
     "BulkAssignmentMessage",
     ["bulk_assignment_id", "coupon_code", "email", "event", "timestamp"],
 )
 
 
+def _make_mailgun_api_request(url):
+    """
+    Makes a Mailgun API request, and retries if the server responds with a 504 (timeout)
+
+    Args:
+        url (str): The URL of the Mailgun API endpoint
+
+    Returns:
+        response (requests.models.Response): The requests library response object
+
+    Raises:
+        requests.exceptions.HTTPError: Raised if the response has a status code indicating an error
+    """
+    resp = requests.get(url)
+    # If there was a timeout (504), retry before giving up
+    tries = 1
+    while (
+        resp.status_code == status.HTTP_504_GATEWAY_TIMEOUT
+        and tries <= MAILGUN_API_TIMEOUT_RETRIES
+    ):
+        tries += 1
+        log.warning(
+            "Mailgun API request timed out (%s). Retrying for attempt %d...", url, tries
+        )
+        resp = requests.get(url)
+    if not resp.ok:
+        resp.raise_for_status()
```

Comment:
You can leave out the `resp.ok` check here. `raise_for_status` will not do anything it `resp.ok` is true.